harvester/terraform-provider-harvester
# Add cpu_pinning and isolate_emulator_thread to vm resource

related to harvester/harvester#6551, include ACC test case for cpu_pinning and isolate_emulator_thread.

### Test Plan
Prerequisite: Prepare a harvester cluster with v1.4 dev version (e.g. [v1.4.0-dev-20240918](https://github.com/harvester/harvester/releases/tag/v1.4.0-dev-20240918))
1. run ACC test
    - Download kubeconfig from harvester, rename it to `kubeconfig_test.yaml` and then move the config under `terraform-provider-harvester/`
    - Execute `make test`
2. terraform apply
    - At least one node enable cpu manager in harvester cluster
    - execute `make` to build binaries
    - if your machine is under amd64, rename bin/terraform-provider-harvester-amd64 to bin/terraform-provider-harvester. Same thing in arm64. 
    - follow https://developer.hashicorp.com/terraform/cli/config/config-file#development-overrides-for-provider-developers to use terraform-harvester-provider from local build.    
      ```tf
      provider_installation {
        dev_overrides {
          "harvester/harvester" = "/home/brandboat/Code/harvester/terraform-provider-harvester/bin"
        }
        direct {}
      }
      ```
    - Add `export TF_CLI_CONFIG_FILE=~/.terraformrc` to `.bashrc`
    - Test terraform create VM https://gist.github.com/brandboat/5a5279d9c13604d2465da3ce0b277743
    - Run `terraform init` and then run `terraform apply`
    - After vm is created, check yaml contains `dedicatedCpuPlacement: true` and `isolateEmulatorThread: true`
